src/main/java/org/truffleruby/parser/ast/SymbolParseNode.java

```diff
@@ -55,14 +55,14 @@
 
     // Interned ident path (e.g. [':', ident]).
     public SymbolParseNode(SourceIndexLength position, String name, Encoding encoding, CodeRange cr) {
-        // The cr argument is ignored since it is not always correct
         super(position);
         this.name = name;  // Assumed all names are already intern'd by lexer.
-        if (encoding.isAsciiCompatible() && StringOperations.isASCIIOnly(name)) {
-            this.rope = StringOperations.encodeRope(name, USASCIIEncoding.INSTANCE, CodeRange.CR_7BIT);
-        } else {
-            this.rope = StringOperations.encodeRope(name, encoding, CodeRange.CR_UNKNOWN);
+
+        if (encoding == USASCIIEncoding.INSTANCE || cr == CodeRange.CR_7BIT) {
+            encoding = USASCIIEncoding.INSTANCE;
         }
+
+        this.rope = StringOperations.encodeRope(name, encoding, cr == null ? CodeRange.CR_UNKNOWN : cr);
```

Comment:
I have to say I prefer the old logic, it's very clear what it does, here we rely very strongly on the lexer CR being correct and we don't verify it, plus we still have to handle a null `cr`.
If the lexer gives us a Rope it would be silly to not take advantage of it, but here we only get CR and we have to re-encode anyway so it seems a tiny gain and a risk.

Reply:
Can we actually have a `null` `cr`?

Reply:
We rely on the lexer CR being correct in a lot of places. It *needs* to be correct if it's going to hand us the CR. And doing it in the lexer saves us extra byte scans. By ignoring the code range, we also bypass the faster String -> Rope conversion for ASCII-only strings. This used to be clearer when we had different calls, but since the `encodeAscii` call was pushed into `encodeRope`, that's hidden a bit.

If you truly don't want to trust the lexer though, then lets do it properly and get rid of the constructor parameter. #735 just left the parameter unused.